sdk/auth/managed_identity_authorizer.go

```diff
@@ -72,7 +73,7 @@ func (a *ManagedIdentityAuthorizer) Token(ctx context.Context, _ *http.Request)
 
 	url := fmt.Sprintf("%s?%s", a.conf.MsiEndpoint, query.Encode())
 
-	body, err := azureMetadata(ctx, url)
+	body, err := retryForIMDS(ctx, url)
```

Comment:
this is already done via the `retryablehttp.Client` [which is instantiated here](https://github.com/hashicorp/go-azure-sdk/blob/4ddf8bbe31e99f92490e553935e836cf2ac81201/sdk/auth/managed_identity_authorizer.go#L172-L180) which [contains some IDMS specific logic](https://github.com/hashicorp/go-azure-sdk/blob/4ddf8bbe31e99f92490e553935e836cf2ac81201/sdk/auth/client.go#L50-L56) - so I believe this change is already present in `main` - is this still an issue when vendoring the latest `main` into the provider?

Reply:
ah, apologies this isn't in main yet: https://github.com/hashicorp/terraform-provider-azurerm/pull/20918

Reply:
Hi @tombuildsstuff, thank you for reviewing.
I found the latest main branch will still cause timeout issue. The reason should be the [timeout ](https://github.com/hashicorp/go-azure-sdk/blob/cacb11e1b5b1f87eb2809ecb9a256cb4a57187c1/sdk/auth/managed_identity_authorizer.go#L178)value is too short. This timeout value is used for all retried http requests, not one of them. After updating the timeout value and the [context deadline](https://github.com/hashicorp/go-azure-sdk/blob/cacb11e1b5b1f87eb2809ecb9a256cb4a57187c1/sdk/auth/managed_identity_authorizer.go#L160), this issue is resolved. Kindly help take another review of this PR. Thanks.

![image](https://user-images.githubusercontent.com/76987228/225207712-3be1f9a2-dbad-4ad3-bf65-e0e334d70735.png)